hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/JetTestSupport.java

```diff
@@ -247,4 +248,36 @@ public void waitForNextSnapshot(JobRepository jr, long jobId, int timeoutSeconds
         }, timeoutSeconds);
         logger.info("Next snapshot found (id=" + snapshotId[0] + ", previous id=" + originalSnapshotId + ")");
     }
+
+    /**
+     * Give this method a job and it will ensure it's no longer running. It
+     * will ignore if it's not running. If the cancellation fails, it will
+     * retry.
+     */
+    public void ditchJob(Job job) {
+        for (;;) {
+            Exception failure;
+            try {
+                job.cancel();
```

Comment:
is there a point to trying to cancel several times? if it gets stuck, we won't be able to get out of it.

Reply:
For example `cancel` fails when the job is restarting or terminating at the very moment. Exception is thrown from `cancel`. After the job starts, you'll be able to cancel. If the job is stuck, `join` won't return, which is what we want: the job that caused the stuck job will fail.

Reply:
Just realized that we'll sleep even in case of a normal job failure. I'll fix that tomorrow...

Reply:
Still if the job gets stuck on cancellation, will it timeout eventually?

Reply:
The entire test will time out, it's 5 mins a think. The cluster is probably in a bad shape anyway and subsequent tests on the same cluster might be affected. But if the test times out after 5 mins, more tests will be run. I probably should shut down the cluster...